experimental/dds/tree2/src/domains/schemaBuilder.ts

```diff
@@ -35,17 +39,155 @@ import {
+				return this.list([types]) as TreeSchema<
+					`${TScope}.${string}`,
+					{
+						structFields: {
+							[""]: FieldSchema<typeof FieldKinds.sequence, NormalizeAllowedTypes<T>>;
+						};
+					}
+				>;
+			} else {
+				assert(Array.isArray(types), "Types should be an array");
+				const names = types.map((t): string => {
+					// Ensure that lazy types (functions) don't slip through here.
+					assert(t instanceof TreeSchema, "invalid type provided");
+					// TypeScript should know `t.name` is a string (from the extends constraint on TreeSchema's name), but the linter objects.
+					// @ts-expect-error: Apparently TypeScript also fails to apply this constraint for some reason and is giving any:
+					type _check = requireFalse<isAny<typeof t.name>>;
+					// Adding `as string` here would silence the linter, but make this code less type safe (since if this became not a string, it would still build).
+					// Thus we explicitly check that this satisfies string.
+					// This would best be done by appending `satisfies string`, but the linter also rejects this.
+					// Therefor introducing a variable to do the same thing as `satisfies string` but such that the linter can understand:
+					const name: string = t.name;
+					// Just incase the compiler and linter really are onto something and this might sometimes not be a string, validate it:
+					assert(typeof name === "string", "Name should be a string");
+					return name;
+				});
+				// Ensure name is order independent
+				names.sort();
+				// Ensure name can't have collisions by quoting and escaping any quotes in the names of types.
+				// Using JSON is a simple way to accomplish this.
+				// The outer `[]` around the result are also needed so that a single type name "Any" would not collide with the "any" case above.
+				inner = JSON.stringify(names);
+			}
+			const fullName = `List<${inner}>`;
+			return getOrCreate(this.structuralTypes, fullName, () =>
+				this.namedList(fullName, nameOrAllowedTypes as T),
+			) as TreeSchema<
+				`${TScope}.${string}`,
+				{
+					structFields: {
+						[""]: FieldSchema<typeof FieldKinds.sequence, NormalizeAllowedTypes<T>>;
+					};
+				}
+			>;
+		}
+		return this.namedList(nameOrAllowedTypes as TName, allowedTypes);
+	}
+
+	/**
+	 * Define (and add to this library) a {@link TreeSchema} for a {@link FieldNode} of a {@link Sequence}.
+	 *
+	 * The name must be unique among all TreeSchema in the the document schema.
+	 *
+	 * @privateRemarks
+	 * TODO: If A custom "List" API is added as a subtype of {@link FieldNode}, this would opt into that.
+	 */
+	private namedList<Name extends TName | string, const T extends ImplicitAllowedTypes>(
+		name: Name,
+		allowedTypes: T,
+	): TreeSchema<
+		`${TScope}.${Name}`,
+		{
+			structFields: {
+				[""]: FieldSchema<typeof FieldKinds.sequence, NormalizeAllowedTypes<T>>;
+			};
+		}
+	> {
+		const schema = new TreeSchema(this, this.scoped(name as TName & Name), {
```

Comment:
what are our options to allow List/Map<scope.Foo> to cast across libraries?

Reply:
They are different schema with different names, so the types being incompatible between libraries is a feature that prevents you from mixing up unrelated types.

If instead you are asking how we could avoid them being different types, the answer is mostly "its complicated" and thus not MVP. The slightly longer version is that if you want two unrelated libraries to return the exact same object as would be required by our current rules, that would have to involve globals , some hard coded scope, implicit single schema builder, and implicit library dependencies. It also likely would work poorly with "any".

Reply:
Thanks. This is an important (and likely common) use case that we want in the future. I filed ADO5938 for it to track post-MVP. @DLehenbauer